web/src/app/app.component.ts

```diff
@@ -0,0 +1,160 @@
+  public errorMessage: string;
+  public contractInstance: any;
+  public contractAddress = '0xc9707E1e496C12f1Fa83AFbbA8735DA697cdBf64';
+  public toDoStatus: any;
+  public inProgressStatus: any;
+  public doneStatus: any;
+  public inactiveButton;
+
+  constructor() {
+    this.toDoStatus = [];
+    this.inProgressStatus = [];
+    this.doneStatus = [];
+
+    etherlime.ContractAt(ToDo, this.contractAddress).then((result) => {
+      this.contractInstance = result;
+      this.successMessage = 'The contract has been set and is ready to interact with it!';
+      this.getToDoStatuses();
+    }).catch((e) => {
+      this.errorMessage = e.message;
+    });
+
+  }
+
+  private async getToDoIndex(_toDo) {
+    const indexCounter = await this.contractInstance.indexCounter();
+    for (let i = 0; i < indexCounter.toNumber(); i++) {
+      let toDo = await this.contractInstance.getToDoByIndex(i);
+      if (toDo === _toDo) {
+        return i
+      }
+    }
+  }
+
+  private async getSignerInstance() {
+    const provider = new ethers.providers.Web3Provider(web3.currentProvider);
+    const signer = await provider.getSigner();
+    const contract = new ethers.Contract(this.contractAddress, ToDo.abi, provider)
+    const instance = await contract.connect(signer);
+    return instance;
+  }
+
+  public async addToDo(todo) {
+    try {
+      this.inactiveButton = true;
+      await this.contractInstance.addToDo(todo.value);
+      this.addInfoMessage('ToDo has been added!');
+      this.toDoStatus.push(todo.value);
+      this.inactiveButton = false;
+    } catch (e) {
+      this.addInfoMessage(`Transaction failed. Are you sure you hadn't already added this ToDo! ${e.message}`);
+      this.inactiveButton = false;
+    }
+  }
+
+  public async assignToDo(assignedTodo) {
+    try {
+      const instance = await this.getSignerInstance();
+      await instance.assignToDo(assignedTodo);
+      this.addInfoMessage('ToDo was assigned to you!')
+    } catch (e) {
+      this.addInfoMessage(`${e.message}. You can not assign to a ToDo if it is not added the list!`)
+    }
+  }
+
+  public async changeToDoStatus(todo, destination) {
+    try {
+      const instance = await this.getSignerInstance();
+      const index = await this.getToDoIndex(todo);
+      await instance.changeToDoStatus(index);
+
+      this.changeCurrentStatus(todo, destination);
+      this.addInfoMessage('Status has been changed!');
+    } catch (e) {
+      this.addInfoMessage(`You can not change ToDo's status if it is not assigned to you! ${e.message}`);
+    }
+  }
+
+  public async removeToDo(todo, destination) {
+    try {
+      const index = await this.getToDoIndex(todo);
+      await this.contractInstance.removeToDo(index);
+      this.cleanCurrentTodo(todo, destination);
+      this.addInfoMessage("ToDo has been removed!")
+    } catch (e) {
+      this.addInfoMessage("ToDo not found to be removed. Add it first!")
+    }
+  }
+
+
+  private async getToDoStatuses() {
```

Comment:
This method will cause a lot of trouble done like this. You should make this method return an array (or object with three arrays) instead of pushing stuff into the `this.toDoStatus` and the other arrays. You are hiding critical piece of logic in this function and noone will guess by reading the name `getToDoStatuses` that this actually not only gets them but also puts them in some global arrays.